cEP-0005.md

```diff
@@ -0,0 +1,157 @@
+- To specify, what analysis should be run the user will not have to care about
+  bears. Instead, an `aspects` setting will be provided allowing the user to
+  specify what aspects of his code should be analyzed. Aspects could be
+  `spelling` or `smells` and are namespaced such that `redundancy.clones` will
+  be possible as well. Aspects can be defined inclusive (analyze those aspects,
+  with the `aspects` settings) or exclusive (analyze all but those aspects,
+  with the `ignore_aspects` setting).
+- Specifying `bears` manually will still be possible as it eases use especially
+  for bear developers.
+
+### Mockup
+
+The following is a mockup of how a configuration could look like.
+
+```ini
+[all]
+# Values can be added to inherited ones only, not within the section because
+# a configuration should describe a state and not involve operations.
+# A system wide coafile can define venv, .git, ... and we would recommend to
+# always use += for ignore even in the default section to inherit those values.
+# += always concatenates with a comma.
+ignore += ./vendor
+
+max_line_length = 80
+
+# This inherits all settings from the `all` section. There is no implicit
+# inheritance.
+[all.python]
+language = Python
+files = **.py
+aspects = smell, redundancy.clone  # Inclusive
+
+[all.c]
+language = C
+files = **.(c|h)
+ignore_aspects = redundancy  # Exclusive
+```
+
+### Bear API
+
+Bears will receive the aspects that should be checked as a parameter and
+results will get metadata to indicate what aspect has been checked.
+
+```python
+from ... import LocalBear
+from ... import Result
+from ...Aspects import Redundancy
+
+
+class RedundancyBear(LocalBear):
+    # coala will run bears based on this metadata
+    LANGUAGES = {"C", "Python"}
+    ASPECTS = {Redundancy.Clone, Redundancy.UnusedImport}
+
+    # Aspects are passed as parameter
+    def run(self, filename, file, aspects):
+        # No documentation required anymore for the bears and their parameters.
+
+        # Check the code ...
+
+        # A bear may yield results for any aspect even if it's not selected but
+        # may use the selected aspects to save performance. coala will filter
+        # out only selected aspects.
+        yield Result.from_values(aspect=Redundancy.Clone, ...)
+```
+
+The concept of aspects allows us to implement a consistency check and reduce
+documentation redundancy. Instead of documenting settings and results in every
+bear, we can create a new aspects class:
+
+```python
+class Redundancy(Aspect):
+    """
+    This meta aspect describes any kind of redundancy in your source code.
+    """
+
+    class Clone(Redundancy):
+        """
+        This redundancy describes a code clone. Code clones are different pieces of
+        code in your codebase that are very similar.
+        """
+
+        MIN_CLONE_TOKEN = ("The number of tokens that have to be equal for it "
+                           "to be detected as a code clone.", int, 20)
```

Comment:
+1000 for this! This is pretty nice and it'll be useful for coala-quickstart too.